docs/Derive.md

```diff
@@ -0,0 +1,107 @@
+# Derive
+
+`derive` is a mechanism that automatically determines how to implement
+interfaces for datatypes based on their members. It also allows you to write
+your own rules for `derive`, called a `deriver`, but most users will likely
+not use that feature.
+
+If you’d like to learn how to `derive` interfaces for your types, read the
+first section of this document. If you’d like to provide a deriver for an
+interface, read the second section of this document.
+
+## I: Using `derive`
+
+In most cases, using `derive` should be as simple as calling it with the type
+name and interface to implement:
+
+```clojure
+(deftype Point [
+  x Int
+  y Int
+])
+(derive Point zero)
+(derive Point =)
+```
+
+The code above will enrich the type `Point` by providing implementations for
```

Comment:
How about:

"The code above will provide implementations of `zero` and `=` for the type `Point` based on its members."